### ISSUE TITLE
How to implement versioning?

### ISSUE THREAD

**Issue by wobbol** [ErgoDox/tmk_ErgoDox#2]
My suggestion is: when we are satisfied with the state of the master branch, point tmk_core sub-module at the most recent release tag of hasu's repo then make a v0.0.1 tag. Then we have another tag named latest-stable that takes a branch of our most recent version tag and commits a sub-module reference of the most recent version tag of tmk_core.

All that might have been hard to follow, but the end goal I want is vX.X.X tags that stay still and latest-stable that is probably managed by a git-hook.

**Comment by wobbol** [comment]
tmk does not do git tag releases. So version-ing will need to be something this org is responsible for.

**Comment by squarefrog** [comment]
So the beauty of using git sub modules is that the commit SHA of our dependencies (in this case tmk_core) is set in stone. That way we can have our own versioning - e.g. V1.0.0 uses tmk_core <sha>. 

Anyone can [update to the latest](http://stackoverflow.com/a/5828396/816002) core if they wish, but the master branch will be more or less set in stone unless a project owner updates and pushes the changes. 

Typically you don't use tag for `latest-stable`. A better approach is to use [`git branch`](https://git-scm.com/book/en/v2/Git-Branching-Basic-Branching-and-Merging). Usually the latest stable code is on `master`, all development happens in `develop`. 

Also consider [semantic versioning](http://semver.org) in conjunction with a [changelog](http://keepachangelog.com/en/0.3.0/). 

Forgive me, if you know the above, but I'm documenting the thought process here for anyone watching the repo 👀

**Comment by wobbol** [comment]
It's fine, when I wrote this issue I was not fully behind the submodule thing and wasn't solid on what it really was.
The thing I was really concerned about is, bumping version _only_ because tmk_core's version bumped. I see that monodevelop team kinda does this and it irritates me bc their git tags are not even remotely stable.
You have sold me on the semantic versioning and changelog although I already knew about semantic versioning

**Comment by wobbol** [comment]
Semantic Versioning requires there be a defined public API. This repos current purpose is a set of config files for tmk_core. There is very little code going on here, so while I support Semantic Versioning, it is not applicable here. if tmk_core used Semantic Versioning, I would say lets make our major version match when we no longer use any deprecated interfaces. But last time I took a look, tmk_core doesn't have a version at all, so we are blowing in the wind here.
 Before the 23rd I plan to have condensed the commits and the closed issues into a set of changes.

**Comment by squarefrog** [comment]
:+1: works for me

**Comment by marknsikora** [comment]
I'm still in the camp that versions are unnecessary effort. The recommendation will always be to use most recent master branch.

**Comment by wobbol** [comment]
@squarefrog seems like you are the only one here who wants this. I know I said I would do it, but I just haven't been able to muster a care enough to do this in light of the discussion here.
If semantic versioning is a killer feature for you please submit a plan for discussion.
changelog only really makes sense when there are stable releases, so I don't see how we can do one without the other.

**Comment by squarefrog** [comment]
i didn't want it - in the first part of my comment above I mention we already set a sub module commit in stone so we're good.
